python/lsst/ts/ess/common/device/base_device.py

```diff
@@ -94,6 +94,19 @@ async def __aexit__(
     ) -> None:
         await self.close()
 
+    def __repr__(self) -> str:
+        return self.__str__()
+
+    def __str__(self) -> str:
+        vars_list = [
+            f"{var}={val!r}"
+            for var, val in vars(self).items()
+            if var not in ["log", "terminator"]
+        ]
+        vars_str = ", ".join(vars_list)
+        st = f"{type(self).__name__}<{vars_str}>"
```

Comment:
If you can construct this with these values as arguments then please consider using () instead if <>.

Reply:
No, those are not constructor arguments.